service/history/handler.go

```diff
@@ -1846,6 +1849,38 @@ func (h *Handler) UpdateWorkflowExecution(
 	return engine.UpdateWorkflowExecution(ctx, request)
 }
 
+func (h *Handler) StreamWorkflowReplicationMessages(
+	server historyservice.HistoryService_StreamWorkflowReplicationMessagesServer,
+) (retError error) {
+	defer log.CapturePanic(h.logger, &retError)
+	h.startWG.Wait()
+
+	if h.isStopped() {
+		return errShuttingDown
+	}
+
+	ctxMetadata, ok := metadata.FromIncomingContext(server.Context())
+	if !ok {
+		return serviceerror.NewInvalidArgument("missing cluster & shard ID metadata")
+	}
+	sourceClusterShardID, targetClusterShardID, err := history.DecodeClusterShardMD(ctxMetadata)
+	if err != nil {
+		return err
+	}
+	if targetClusterShardID.ClusterName != h.clusterMetadata.GetCurrentClusterName() {
```

Comment:
sourceClusterShardID.ClusterName?

Reply:
this is checking if target cluster is local cluster